### PR TITLE
Document the removal of AbstractSearchEngine::parseSearchQuery()

### PR DESCRIPTION
**Pull request by TimWolla**

### CODE REVIEW

File: pages/migration/wsc-53/migration_wsc-53_php.md [hunk 1]
```diff
@@ -75,3 +75,13 @@ The reCAPTCHA v1 endpoints have already been turned off by Google and always ret
 Thus the implementation was completely non-functional even before this change.
 
 See [WoltLab/WCF#3781](https://github.com/WoltLab/WCF/pull/3781) for details.
+
+## Search
+
+The generic implementation in the `AbstractSearchEngine::parseSearchQuery()` method was dangerous, because it did not have knowledge about the search engine’s specifics.
+The implementation was completely removed `parseSearchQuery()` now always throws a `\BadMethodCallException`.
```
Comment (Gravatronics) [inline pages/migration/wsc-53/migration_wsc-53_php.md:82]:
```suggestion
The implementation was completely removed: `AbstractSearchEngine::parseSearchQuery()` now always throws a `\BadMethodCallException`.
```

File: pages/migration/wsc-53/migration_wsc-53_php.md [hunk 2]
```diff
@@ -75,3 +75,13 @@ The reCAPTCHA v1 endpoints have already been turned off by Google and always ret
...
+
+If you implemented a custom search engine and relied on this method you can inline the previous implementation to preserve existing behavior.
```
Comment (Gravatronics) [inline pages/migration/wsc-53/migration_wsc-53_php.md:84]:
```suggestion
If you implemented a custom search engine and relied on this method, you can inline the previous implementation to preserve existing behavior.
```

File: pages/migration/wsc-53/migration_wsc-53_php.md [hunk 3]
```diff
@@ -75,3 +75,13 @@ The reCAPTCHA v1 endpoints have already been turned off by Google and always ret
...
+You should take the time to verify the results against the manual of the search engine to make sure it cannot generate malformed queries or security issues.
```
Comment (Gravatronics) [inline pages/migration/wsc-53/migration_wsc-53_php.md:85]: "results" is too vague here and should be replaced with a more specific word/phrase.

**Review by Gravatronics — CHANGES_REQUESTED** [review]

File: pages/migration/wsc-53/migration_wsc-53_php.md (hunk 3 above)
Comment (TimWolla) [reply pages/migration/wsc-53/migration_wsc-53_php.md:85]: How about this:

```suggestion
You should take the time to verify the generated queries against the manual of the search engine to make sure it cannot generate malformed queries or security issues.
```

Comment (TimWolla) [reply pages/migration/wsc-53/migration_wsc-53_php.md:85]: Did s/generated/rewritten/ in my suggestion.

Comment (Gravatronics) [reply pages/migration/wsc-53/migration_wsc-53_php.md:85]: Yes, that is better!

**Pull request merged**